File: backend/internal/pkg/dto/modifier.go
package dto

const (
	ModifierTypeConstantOffset      = "MODIFIER_TYPE_CONSTANT_OFFSET"
	ModifierTypeHysteresis          = "MODIFIER_TYPE_HYSTERESIS"
	ModifierTypeInertia             = "MODIFIER_TYPE_INERTIA"
	ModifierTypeNonLinearDependence = "MODIFIER_TYPE_NONLINEAR_DEPENDENCE"
	ModifierTypeProgressingOffset   = "MODIFIER_TYPE_PROGRESSING_OFFSET"
	ModifierTypeQuantization        = "MODIFIER_TYPE_QUANTIZATION"
	ModifierTypeRandomAddDash       = "MODIFIER_TYPE_RANDOM_ADD_DASH"
	ModifierTypeRandomFixedDash     = "MODIFIER_TYPE_RANDOM_FIXED_DASH"
	ModifierTypeWhiteNoise          = "MODIFIER_TYPE_WHITE_NOISE"
	ModifierTypeDependence          = "MODIFIER_TYPE_DEPENDENCE"
)

type Modifier struct {
	Type string      `json:"type"`
	Data interface{} `json:"data"`
}

type ConstantOffsetModifier struct {
	Offset float64 `json:"offset"`
}

type HysteresisModifier struct {
	Percentage int `json:"percentage"`
}

type InertitaModifier struct {
	Value  float64  `json:"value"`
	Period Duration `json:"period"`
}

type NonLinearDependenceModifier struct {
	Coefficient float64 `json:"coefficient"`
	Center      float64 `json:"center"`
}

type ProgressingOffsetModifier struct {
	Value    float64  `json:"value"`
	Interval Duration `json:"interval"`
}

type QuantizationModifier struct {
	Quant float64 `json:"quant"`
}

type RandomAddDashModifier struct {
	Generator       Prng     `json:"generator"`
	MinAddValue     float64  `json:"min_add_value"`
	MaxAddValue     float64  `json:"max_add_value"`
	MinDashDuration Duration `json:"min_dash_duration"`
	MaxDashDuration Duration `json:"max_dash_duration"`
	AvgPeriod       Duration `json:"avg_period"`
}

type RandomFixedDashModifier struct {
	Generator       Prng     `json:"generator"`
	Value           float64  `json:"value"`
	MinDashDuration Duration `json:"min_dash_duration"`
	MaxDashDuration Duration `json:"max_dash_duration"`
	AvgPeriod       Duration `json:"avg_period"`
}

type WhiteNoiseModifier struct {
	Generator Prng    `json:"generator"`
	MaxValue  float64 `json:"max_value"`
}

type DependenceModifier struct {
	SimulatorName string  `json:"simulator_name"`
	Center        float64 `json:"center"`
	Coefficient   float64 `json:"coefficient"`
}

File: frontend/src/constants/modifier.js
export const MODIFIER_TO_NAME = {
    "MODIFIER_TYPE_CONSTANT_OFFSET": "Сдвиг",
    "MODIFIER_TYPE_HYSTERESIS": "Гистерезис",
    "MODIFIER_TYPE_INERTIA": "Инертность",
    "MODIFIER_TYPE_NONLINEAR_DEPENDENCE": "Нелинейность",
    "MODIFIER_TYPE_PROGRESSING_OFFSET": "Прогрессирующий сдвиг",
    "MODIFIER_TYPE_QUANTIZATION": "Квантование",
    "MODIFIER_TYPE_RANDOM_ADD_DASH": "Случайный добавочный сдвиг",
    "MODIFIER_TYPE_RANDOM_FIXED_DASH": "Случайный фиксированный сдвиг",
    "MODIFIER_TYPE_WHITE_NOISE": "Белый шум",
    "MODIFIER_TYPE_DEPENDENCE": "Зависимость",
}

export const MODIFIER_DATA_TO_DATA_NAME = {
    "MODIFIER_TYPE_CONSTANT_OFFSET": {
        "offset": "Значение"
    },
    "MODIFIER_TYPE_HYSTERESIS": {
        "percentage": "Процент"
    },
    "MODIFIER_TYPE_INERTIA": {
        "value": "Макс. изменение за период",
        "period": "Период"
    },
    "MODIFIER_TYPE_NONLINEAR_DEPENDENCE": {
        "coefficient": "Коэффициент",
        "center": "Центровая точка"
    },
    "MODIFIER_TYPE_PROGRESSING_OFFSET": {
        "value": "Изменение за перод",
        "interval": "Период"
    },
    "MODIFIER_TYPE_QUANTIZATION": {
        "quant": "Квант"
    },
    "MODIFIER_TYPE_RANDOM_ADD_DASH": {
        "min_add_value": "Мин. добавочное значение",
        "max_add_value": "Макс. добавочное значение",
        "min_dash_duration": "Мин. длительность сдвига",
        "max_dash_duration": "Макс. длительность сдвига",
        "avg_period": "Период возникновения сдвига"
    },
    "MODIFIER_TYPE_RANDOM_FIXED_DASH": {
        "value": "Значение сдвига",
        "min_dash_duration": "Мин. длительность сдвига",
        "max_dash_duration": "Макс. длительность сдвига",
        "avg_period": "Период возникновения сдвига"
    },
    "MODIFIER_TYPE_WHITE_NOISE": {
        "max_value": "Максимальное значение"
    },
    "MODIFIER_TYPE_DEPENDENCE": {
        "simulator_name": "Зависим от", 
        "center": "Центровая точка",
        "coefficient": "Коэффициент"
    },
}
